data_prep/build_bridge_timeseries_june2018.R

```diff
@@ -111,16 +124,19 @@ interpolateMissingValsWLastSeen <- function(vec){
   return(out)
 }
 
+
 bridge_df_by_bridge_and_year <- full_bridge_df_sm %>%
   group_by(bridgeID, data_year) %>%
   summarize_all(funs(getMaxAdjusted(.))) %>%
   arrange(bridgeID, data_year) %>%
   ungroup()
 
+checkNAs(bridge_df_by_bridge_and_year)
+
+
 bridge_df_by_bridge_and_year <- bridge_df_by_bridge_and_year %>%
   arrange(bridgeID, data_year) %>%
   ungroup() %>%
-  group_by(bridgeID) %>%
```

Comment:
This `group_by(bridgeID)` triggered the following error:

`Error: Column 'bridgeID' can't be modified because it's a grouping variable`

Reply:
I'm not sure that the differences in datasets are innocuous. 

Here is a quick test I'm doing right after the "merge data" header. I see a difference in "percent_adt_truck_109" and in "num_obs".

We should make sure we are 100% confident about what is happening before we move on.

```
load("data/bridge_timeseries_june4.rdata")
bridge_ts <- bridge_ts

bridge_ts <- bridge_df_by_bridge_and_year %>% ## Note: This no longer has every year represented
  left_join(bridge_spending_by_bridge_and_year, by=c("bridgeID","data_year")) %>%
  arrange(bridgeID, data_year)

bridge_ts <- bridge_ts %>%
  arrange(bridgeID, data_year) %>%
  mutate(
    spending = ifelse(is.na(total_bridge_spending), 0, total_bridge_spending ),
    spending_in_year = ifelse( spending > 0, calendar_year, NA),
    project_end_year = ifelse(spending > 0, last_end_year, NA),
    project_init_year = ifelse(spending > 0, first_start_year, NA)
  )

newdf <- data.frame(source = rep("new", nrow(bridge_ts))) %>% bind_cols(bridge_ts)
olddf <- data.frame(source = rep("old", nrow(bridge_ts))) %>% bind_cols(test)
t <- newdf %>% select(-num_obs) %>% anti_join(olddf, by = c("bridgeID", "data_year", "bin", "cndtn", "area_sf", "adt_029", "future_adt_114", "percent_adt_truck_109", "structure_len_mt_049", "appr_width_mt_032", "median_code_033", "roadway_width_mt_051", "traffic_lanes_on_028a", "degrees_skew_034", "structure_flared_035", "open_closed_posted_041", "structure_kind_043a", "structure_type_043b", "appr_rail_036c", "appr_rail_end_036d", "railings_036a", "left_curb_mt_050a", "right_curb_mt_050b", "deck", "superstructure", "substructure", "design_load_031", "year_built_latest", "age", "start_year", "num_projects", "total_bridge_spending", "avg_bridge_spending", "first_start_year", "last_end_year", "calendar_year", "proj_no", "spending", "spending_in_year", "project_end_year", "project_init_year"))
a <- olddf %>% filter(bridgeID=="A02006")
d <- bind_rows(a[1,],t[1,])
```

Reply:
We should also try not to keep deprecated syntax. Please update the code vis a vis this warning:
```
Warning message:
funs() is soft deprecated as of dplyr 0.8.0
Please use a list of either functions or lambdas: 

  # Simple named list: 
  list(mean = mean, median = median)

  # Auto named with `tibble::lst()`: 
  tibble::lst(mean, median)

  # Using lambdas
  list(~ mean(., trim = .2), ~ median(., na.rm = TRUE))
```

Reply:
I'll change funs() to one of the supproted alternatives. My rational for keeping funs() was that it was so much faster than the alternatives. My bad!

I also think I caught why differences in "percent_adt_truck_109" and in "num_obs" are showing up. When I deleted `group_by(bridgeID) `, my intent was to eliminate grouping when the values are interpolated with `mutate_at`. BUT this also meant that when `mutate` is used to create "percent_adt_truck_109" and in "num_obs" , the observations weren't grouped! After  grouping them, I am getting the same data set, with exactly 68,609 observations.